src/data/fetchProfessor.ts

```diff
@@ -0,0 +1,52 @@
+
+import fetch from "node-fetch";
+import type { fetchProfessor_params, professorData } from "./nebulaInterfaces.js";
+import { unRegister } from "./removeLater.js";
+
+// TO RUN, RUN THIS WITH 
+// ts-node-esm fetchProfessor.ts
+
+let NEBULA_API_KEY = "EM~eW}G<}4qx41fp{H=I]OZ5MF6T:1x{<GF:~v<";
+
+// Helper function to fetch data from Nebula API. Returns a promise.
+function fetchHelper(paramsObj: fetchProfessor_params) {
+    const headers = {
+        "x-api-key": unRegister(NEBULA_API_KEY),
+        Accept: "application/json",
+    };
+    const getDataPromise = new Promise(async (resolve, reject) => {
+        try {
+            const res = await fetch(
+                `https://api.utdnebula.com/professor?first_name=${paramsObj.firstName}&last_name=${paramsObj.lastName}`,
+                {
+                    method: "GET",
+                    headers: headers,
+                }
```

Comment:
This can be put in a single object scoped globally to this file since it will be the same across all Nebula API fetch functions

```js
const fetchOptions = {
    method: "GET",
    headers: {
        "x-api-key": unRegister(NEBULA_API_KEY),
        Accept: "application/json",
    },
}
```